lib/ansible/modules/cloud/openstack/os_server_facts.py

```diff
@@ -63,14 +70,16 @@ def main():
     argument_spec = openstack_full_argument_spec(
         server=dict(required=False),
         detailed=dict(required=False, type='bool'),
+        all_projects=dict(required=False, type='bool'),
```

Comment:
Should this have default=False?

Reply:
I copied it from "detailed" from above, which does not have 'default=False', but does default to false.  Should I add it to both?

Reply:
default will be 'None' which is 'falsey', that should be good enough and even allow for smarter logic when it was 'unset'

Reply:
<nod> But the default is documented as "no" and the code is passing on None to the cloud.list_servers() API rather than False.  So I think that this probably should be default=False.  (detailed probably should be as well).

If we are doing something that we need to know whether the user passed in the value or not, then default=None makes sense.  But then our code would have some processing which looked at whether it was None and did something different based on that (for instance, omitting the parameter to cloud.list_servers altogether).

Reply:
Ok, I'll make those changes. Thanks for the review!

Reply:
I looked through `openstacksdk` and it seems it is expecting `True` or `False`. Defaulting to `False` is a much safer bet than defaulting to `None`.

Reply:
I've pushed a commit that implements default=False.  Does that look like what you'd expect?

Reply:
Yup, that looks correct.